index.js

```diff
@@ -22,12 +22,21 @@ const addBundleVisualizer = (options = {}, behindFlag = false) => config => {
   return config;
 };
 
-const getBabelLoader = config => {
-  const babelLoaderFilter = rule =>
-    rule.loader &&
-    rule.loader.includes("babel") &&
-    rule.options &&
-    rule.options.plugins;
+const getBabelLoader = (config, isOutsideOfApp) => {
+  let babelLoaderFilter;
+  if (isOutsideOfApp) {
+    babelLoaderFilter = rule =>
+      rule.loader &&
+      rule.loader.includes("babel") &&
+      rule.options &&
+      !rule.options.plugins;
```

Comment:
The problem with this check is that once you call `addBabelPluginOutSideOfApp` or `addBabelPluginsOutSideOfApp`, we can no longer find the "outside" `babel-loader`, because `rule.options.plugins` exists. We also can't differentiate between the app loader and the "outside" loader anymore, because according to the filter rules, they're identical.

I think it makes more sense to find all rules that include `babel` and then differentiating based on index. `0` will always be the app loader and `1` will always be the "outside" loader.

Does that sound correct?

Reply:
Yes.
Actually, I thought this problem at first.But I thought the 'customize-cra' would only execute once.Today i reconsidered it, `addBabelPluginsOutSideOfApp` will be called more than a time.
So it is really a mistake.
I will fix it.

But are you sure the index `1` is always be the app loader and the `1` will always be the "outside" loader ? 
Is it a better idea that differentiate them by `include` or `exclude` option?

Reply:
I made a diff of the app config (left) and the outside config (right) from [react-scripts/config/webpack.config.js#L354-L442](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/webpack.config.js#L354-L442): https://www.diffchecker.com/SXZw2sjY

So `include`/`exclude` would work fine. You could also check `sourceMaps`, which is always false in outside config.

Also just a note: I rearranged some of the files in the project (see #118). `getBabelLoader` is now in `src/utilities` and other babel-related functions are in `src/customizers/babel`.

Reply:
OK.
It is really a nice job to add UnitTest to project !!
I will open a new pull request with test suits and new snapshots.